pkg/controller/networkpolicy/validate.go

```diff
@@ -462,9 +462,35 @@ func (v *antreaPolicyValidator) validateAppliedTo(ingress, egress []crdv1alpha1.
 	if numAppliedToInRules > 0 && (numAppliedToInRules != len(ingress)+len(egress)) {
 		return "appliedTo field should either be set in all rules or in none of them", false
 	}
-	for _, eachSpecAppliedTo := range specAppliedTo {
-		if eachSpecAppliedTo.Group != "" && numFieldsSetInPeer(eachSpecAppliedTo) > 1 {
-			return "group cannot be set with other peers in appliedTo", false
+
+	checkAppliedTo := func(appliedTo []crdv1alpha1.NetworkPolicyPeer) (string, bool) {
+		for _, eachAppliedTo := range appliedTo {
+			appliedToFieldsNum := numFieldsSetInPeer(eachAppliedTo)
+			if eachAppliedTo.Group != "" && appliedToFieldsNum > 1 {
+				return "group cannot be set with other peers in appliedTo", false
+			}
+			if eachAppliedTo.ServiceAccount != nil && appliedToFieldsNum > 1 {
+				return "serviceAccounts cannot be set with other peers in appliedTo", false
+			}
+		}
+		return "", true
+	}
+
+	reason, allowed := checkAppliedTo(specAppliedTo)
+	if !allowed {
+		return reason, allowed
+	}
+
+	for _, eachIngress := range ingress {
```

Comment:
Does it also fix the bug that appliedTo in rule was not validated before?

Reply:
Yes.